src/vm/methodtable.cpp

```diff
@@ -7006,31 +7042,53 @@ BOOL MethodTable::FindDefaultMethod(
                 {
                     if (pCurMT->HasSameTypeDefAs(pInterfaceMT))
                     {
-                        // Generic variance match
+                        // Generic variance match - we'll instantiate pCurMD with the right type arguments later
                         pCurMD = pInterfaceMD;
                     }
                     else
                     {
                         //
-                        // Parent interface - search for an methodimpl for explicit override
+                        // A more specific interface - search for an methodimpl for explicit override
                         // Implicit override in default interface methods are not allowed
                         //
                         MethodIterator methodIt(pCurMT);
                         for (; methodIt.IsValid(); methodIt.Next())
                         {
                             MethodDesc *pMD = methodIt.GetMethodDesc();
-                            if (pMD->IsVirtual() && !pMD->IsAbstract() && pMD->IsMethodImpl())
+                            int targetSlot = pInterfaceMD->GetSlot();
+
+                            if (pMD->IsMethodImpl())
                             {
                                 MethodImpl::Iterator it(pMD);
-                                while (it.IsValid())
+                                for (; it.IsValid(); it.Next())
                                 {
-                                    if (it.GetMethodDesc() == pInterfaceMD)
+                                    MethodDesc *pDeclMD = it.GetMethodDesc();
+
+                                    if (pDeclMD->GetSlot() != targetSlot)
+                                        continue;
+
+                                    MethodTable *pDeclMT = pDeclMD->GetMethodTable();
+                                    BOOL bMatch = FALSE;
+                                    if (pDeclMT->ContainsGenericVariables())
                                     {
+                                        TypeHandle thInstDeclMT = ClassLoader::LoadGenericInstantiationThrowing(
+                                            pDeclMT->GetModule(),
+                                            pDeclMT->GetCl(),
+                                            pCurMT->GetInstantiation());
+                                        MethodTable *pInstDeclMT = thInstDeclMT.GetMethodTable();
+                                        if (pInstDeclMT == pInterfaceMT)
+                                        {
+                                            // This is a matching override. We'll instantiate pCurMD later
+                                            pCurMD = pMD;
+                                            break;
```

Comment:
>break [](start = 44, length = 5)

This will just break from the inner loop. I think you also want to break out of the outer loop